lambdas/personInfo/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/PersonInfoHandler.java

```diff
@@ -0,0 +1,188 @@
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input,
+            Context context) { // need something that returns a custom response
+
+        try {
+            LOGGER.info(
+                    "Initiating lambda {} version {}",
+                    context.getFunctionName(),
+                    context.getFunctionVersion());
+
+            // Recorded here as sending metrics during function init may fail depending on lambda
+            // config
+            if (!functionInitMetricCaptured) {
+                eventProbe.counterMetric(
+                        Definitions.LAMBDA_PERSON_INFO_FUNCTION_INIT_DURATION,
+                        functionInitMetricLatchedValue);
+                LOGGER.info("Lambda function init duration {}ms", functionInitMetricLatchedValue);
+                functionInitMetricCaptured = true;
+            }
+
+            // Lambda Lifetime
+            long runTimeDuration =
+                    System.currentTimeMillis() - FUNCTION_INIT_START_TIME_MILLISECONDS;
+            Duration duration = Duration.of(runTimeDuration, ChronoUnit.MILLIS);
+            String formattedDuration =
+                    String.format(
+                            "%d:%02d:%02d",
+                            duration.toHours(), duration.toMinutesPart(), duration.toSecondsPart());
+            LOGGER.info(
+                    "Lambda {}, Lifetime duration {}, {}ms",
+                    context.getFunctionName(),
+                    formattedDuration,
+                    runTimeDuration);
+
+            LOGGER.info("Handling Person Info request");
+            Map<String, String> headers = input.getHeaders();
+            final String sessionId = headers.get("session_id");
+
+            // QuerySessionTable to get sessionItem
+            SessionItem sessionItem = sessionService.getSession(sessionId);
+
+            // throw error if session is not found
+            if (sessionItem == null || sessionItem.getSessionId() == null) {
+                throw new SessionNotFoundException("Session is not found");
+            }
+
+            // Get context from sessionItem
+            String coreContext = sessionItem.getContext();
+            LOGGER.info("Context Value = {}", coreContext);
+
+            APIGatewayProxyResponseEvent emptyResponse = new APIGatewayProxyResponseEvent();
+            emptyResponse.setStatusCode(HttpStatusCode.NO_CONTENT);
+
+            if (coreContext != null) {
+                if (coreContext.equalsIgnoreCase("check_details")) {
+                    ObjectMapper objectMapper =
+                            new ObjectMapper().registerModule(new JavaTimeModule());
+
+                    LOGGER.info("Querying PersonIdentity table with sessionId...");
+                    LOGGER.info("SessionIDUUID = {}", UUID.fromString(sessionId));
+                    var personIdentityDetailed =
```

Comment:
We need to handle what happens if the person identity is not found.

Throw an oauth invalid_request